internal/checker/checker.go

```diff
@@ -30379,3 +30407,26 @@ func (c *Checker) GetEmitResolver(file *ast.SourceFile, skipDiagnostics bool) *e
 func (c *Checker) GetAliasedSymbol(symbol *ast.Symbol) *ast.Symbol {
 	return c.resolveAlias(symbol)
 }
+
+func (c *Checker) sourceFileMayBeEmitted(sourceFile *ast.SourceFile) bool {
```

Comment:
This function already existed in emitter.go; it may just need to be moved (let's go with `internal/outputpaths`) and remove its unused `host` parameter for now.